File: 250617/숫자 차례로 출력하기/to-output-numerically.java
import java.util.Scanner;
public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        // Please write your code here.

        oneToN(n);
        System.out.println();
        nToOne(n);
    }

    public static void oneToN(int n) {
        if (n == 0) return;

        oneToN(n - 1);
        System.out.print(n + " ");
    } 

    public static void nToOne(int n) {
        if (n == 0) return;

        System.out.print(n + " ");
        nToOne(n - 1);
    }
}

File: 250923/숫자 차례로 출력하기/to-output-numerically.py
n = int(input())

# Please write your code here.

def p1(n) :
    if n == 0 :
        return
    
    p1(n-1)
    print(n,end=" ")

def p2(n) :
    if n == 0 :
        return

    print(n,end=" ")
    p2(n-1)

p1(n)
print()
p2(n)